### DISCUSSION TITLE
Impact build is getting stuck in *** Creating SearchPathImpactFinder

### DISCUSSION THREAD

**Discussion by ratnamGT** [IBM/dbb-zappbuild#580] [Q&A]
Hi All,

I am trying to run impact build with Mainframe USS.I am using DBB2.0 and zAppbuild 3. first I ran scan to create metadata then ran the impact build for one program and it executed successfully but when I run the impact build second time it is getting stuck every time in below step.Also When I tried to change copybook and run impact build again then it is getting stuck . Could you please help me on this?

** Storing 2 logical files in MetadataStore collection 'MortgageApplication-main'
*** Perform impacted analysis for changed files.
** Found build script mapping for MortgageApplication/cobol/hello.cbl. Adding to build list
** Performing impact analysis on changed file MortgageApplication/cobol/hello.cbl
*** Creating SearchPathImpactFinder with collections [MortgageApplication-main, MortgageApplication-main-outputs] and impactSearch configuration search:/u/gmsratn/workspace/?path=MortgageApplication/copybook/*.cpysearch:/u/gmsratn/workspace/?path=MortgageApplication/bms/*.bmssearch:[:LINK]/u/gmsratn/workspace/?path=MortgageApplication/cobol/*.cbl

**Comment by dennis-behm** [comment]
@ratnamGT can you please list which exact version of DBB you are using? Are you using one of the the DBB daemons?

It does not go beyond this point?

**Reply by ratnamGT** [reply]
Hi @dennis-behm  , I am using DBB 2.0 and yes I am using DBB_PERSONAL_DEAMON to run the script. Sometimes I am also getting the below error ,however I have setup DBB env variables
**BGZTK0017E An error occurred when loading DBB_HOME environment variable. Variable is not set.**

**Reply by dennis-behm** [reply]
Intermitted issues are difficult to diagnose.

Which exact level of DBB 2.0 are you running on?
https://www.ibm.com/support/pages/fix-list-ibm-dependency-based-build

What happens if you run without the daemon?

**Reply by ratnamGT** [reply]
@dennis-behm How can I find the level number in USS? I am not sure about it . Could you please guide me to find the level of DBB by checking in DBB home directory?
We will fix DBB_HOME issue .It might be due to profile file. Actual issue is in the impact build is getting stuck.Please help us on this

**Reply by dennis-behm** [reply]
zAppBuild verbose mode actually prints the version that are using:  
```
** Running zAppBuild Version 3.10.1-BASE 
** Running DBB Toolkit Version 3.0.0.1 25-Nov-2024 15:33:38 
```
Alternatively use the validation.groovy that is located in the dbb install directory on USS

**Reply by ratnamGT** [reply]
@dennis-behm , Please find the details below.

Info about this DBB installation
--------------------------------
DBB Version: 2.0.0
Build Number: 63
Release Date: 28-Sep-2022 17:41:54
I am using zAppbuild 3.10

**Reply by dennis-behm** [reply]
I pasted the link to the fixes earlier in the thread. I recommend updating to at least 2.0.2.1, you might even go ahead and move to 3.0.

Are you using the file metadata store or the db2 metadatastore? If it is db2, please consider the required schema updates - it is mentioned in the N&N - https://www.ibm.com/docs/en/dbb/2.0?topic=overview-what-is-new-noteworthy

**Comment by ratnamGT** [comment]
@dennis-behm Thanks for the response . Could you please suggest the fix number for this issue  ? We do not have plan as of now to upgrade DBB. We are using DBB file metadata store. Could you please suggest the work around to fix this issue?

**Reply by dennis-behm** [reply]
I have not seen this behaviour so far. So, there is no guarantee that an update will solve it. 

Have you tried the scenarios that I was proposing? What is the behaviour without the daemon?

**Reply by ratnamGT** [reply]
@dennis-behm The same behavior ,I am getting while using without personal daemon also.Impact build is getting stuck in the same step

** Storing 1 logical files in MetadataStore collection 'MortgageApplication-main'
/u/adcdmst>
*** Perform impacted analysis for changed files.
** Performing impact analysis on changed file MortgageApplication/copybook/epsmlis.cpy
*** Creating SearchPathImpactFinder with collections [MortgageApplication-main, MortgageApplication-main-outputs] and impactSearch configuration search:/u/gmszfs/workspace/?path=MortgageApplication/copybook/*.cpysearch:/u/gmszfs/workspace/?path=MortgageApplication/bms/*.bmssearch:[:LINK]/u/gmszfs/workspace/?path=MortgageApplication/cobol/*.cbl

**Reply by dennis-behm** [reply]
Please update to the latest DBB 2.0 release. 

If you don't have access to ShopZ or PassportAdvantage, you find it on this page:
https://ibm.github.io/mainframe-downloads/products/ibm-dependency-based-build.html

You can update the link of the trial download and point to 2.0.2.1: 
https://www14.software.ibm.com/cgi-bin/weblap/lap.pl?popup=Y&li_formnum=L-DGNN-L3MFDK&accepted_url=https://public.dhe.ibm.com/ibmdl/export/pub/software/htp/zos/tools/dbb/toolkit/dbb-ztoolkit-trial-2.0.2.1.tar
